extract_token_from_browser.py

```diff
@@ -1,64 +1,176 @@
+    if args.browser in ["firefox", "firefox-snap"]:
+        browser_data = Path.home().joinpath(
+            "Library/Application Support/Firefox/Profiles")
+    else:
+        browser_data = Path.home().joinpath(
+            "Library/Application Support/Chrome")
 else:
-    print("Currently only Linux and macOS is supported by this script", file=sys.stderr)
+    print("Currently only Linux and macOS is supported by this script",
+          file=sys.stderr)
     sys.exit(1)
 
+cookies_path = None
+profile = args.profile
+if args.browser in ["firefox", "firefox-snap"]:
+    cookie_d_query = (
+        "SELECT value FROM moz_cookies WHERE host = '.slack.com' "
+        "AND name = 'd'"
+    )
+    cookie_ds_query = (
+        "SELECT value FROM moz_cookies WHERE host = '.slack.com' "
+        "AND name = 'd-s'"
+    )
+    if not profile:
+        profile = "*.default*"
+
+    default_profile_path = max(
+        [ x for x in browser_data.glob(profile)], key=os.path.getctime)
+    if not default_profile_path:
+        print("Couldn't find the default profile for Firefox", file=sys.stderr)
+        sys.exit(1)
+else:
+    # b'v10' is for Chromium, but not Chrome, it seems?
+    prefix =  b'v11'
+    cookie_d_query = (
+        "SELECT encrypted_value FROM cookies WHERE "
+        "host_key = '.slack.com' AND name = 'd'"
+    )
+    cookie_ds_query = (
+        "SELECT encrypted_value FROM cookies WHERE "
+        "host_key = '.slack.com' AND name = 'd-s'"
+    )
+    if not profile:
+        profile = "Default"
+
+    default_profile_path = browser_data.joinpath(profile)
+    cookies_path = default_profile_path.joinpath("Cookies")
+
+if not cookies_path:
+    cookies_path = default_profile_path.joinpath("cookies.sqlite")
+
+cookie_d_value = None
+cookie_ds_value = None
 try:
-    default_profile_path = next(firefox_path.glob("*.default-release"))
-except StopIteration:
-    print("Couldn't find the default profile for Firefox", file=sys.stderr)
-    sys.exit(1)
+    con = sqlite3.connect(f"file:{cookies_path}?immutable=1", uri=True)
+    cookie_d_value = con.execute(cookie_d_query).fetchone()[0]
+    cookie_ds_value = con.execute(cookie_ds_query).fetchone()[0]
+except TypeError:
+    if not cookie_d_value:
+        print("Couldn't find the 'd' cookie value", file=sys.stderr)
+        sys.exit(1)
+    pass
+finally:
+    con.close()
 
-cookies_path = default_profile_path.joinpath("cookies.sqlite")
-con = sqlite3.connect(f"file:{cookies_path}?immutable=1", uri=True)
-cookie_d_query = (
-    "SELECT value FROM moz_cookies WHERE host = '.slack.com' AND name = 'd'"
-)
-cookie_d_value = con.execute(cookie_d_query).fetchone()[0]
-cookie_ds_query = (
-    "SELECT value FROM moz_cookies WHERE host = '.slack.com' AND name = 'd-s'"
-)
-cookie_ds_values = con.execute(cookie_ds_query).fetchone()
-con.close()
+if args.browser in ["chrome", "chrome-beta"]:
+    bus = secretstorage.dbus_init()
+    collection = secretstorage.get_default_collection(bus)
```

Comment:
I don't have a service providing `org.freedesktop.secrets` so this fails for me with this message:

```
Traceback (most recent call last):
  File "/usr/lib/python3.10/site-packages/secretstorage/util.py", line 48, in send_and_get_reply
    raise DBusErrorResponse(resp_msg)
jeepney.wrappers.DBusErrorResponse: [org.freedesktop.DBus.Error.ServiceUnknown] ('The name org.freedesktop.secrets was not provided by any .service files',)

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/home/trygve/dev/wee-slack/./extract_token_from_browser.py", l
ine 130, in <module>
    collection = secretstorage.get_default_collection(bus)
  File "/usr/lib/python3.10/site-packages/secretstorage/collection.py", line 177, in get_default_collection
    return Collection(connection)
  File "/usr/lib/python3.10/site-packages/secretstorage/collection.py", line 45, in __init__
    self._collection.get_property('Label')
  File "/usr/lib/python3.10/site-packages/secretstorage/util.py", line 67, in get_property
    (signature, value), = self.send_and_get_reply(msg)
  File "/usr/lib/python3.10/site-packages/secretstorage/util.py", line 58, in send_and_get_reply
    raise SecretServiceNotAvailableException(data) from resp
secretstorage.exceptions.SecretServiceNotAvailableException: The name org.freedesktop.secrets was not provided by any .service files
```

Preferably, this should be handled gracefully, and we should try `peanuts` as the default password (as that's what Chrome uses by default).